General/Glossary.md

```diff
@@ -2,38 +2,64 @@
 In Pharo, there is some specific vocabulary that is used inside the community to designate concepts provided by the language.
 This page aims to provide disambiguation for the words belonging to this vocabulary.
 
+## Binary message
+A binary message is a [message](#message) composed of special character(s) where two objects are involved (the [receiver](#receiver) and one argument), which is the reason it is call binary. This is mostly used for arithmetic, comparison, and logical operations. For example, `+` is a binary message involving the receiver and the argument in parameter.
+
+## Bootstrap
+
 ## Browser
 The browser designate the tool for browsing and editing packages, classes and methods. In Pharo 6.1 and greater, the browser is Calypso.
 
 ## Candidates
 In the context of a method-call the candidates are the potential classes in the system that can receive the call. This list is computed from a static analysis of the source code.
 
+## Changes (file)
+The changesfile logs of all source code modifications (especially all thechanges you did while programming) since the [sources file](#sources) was generated. This facilitates a per method history for diffs or reverting. It means that even if you did not manage to save the image file on a crash or you just forgot, you can recover your changes from this file. A changes file is always coupled with a image file. They work in pair.
+
+> Note: Since Pharo 5 Pharo contains a project called Epicea implementing a new logging system. The long term goal of Epicea is to replace the changes file, but we are not yet at this stage.
+
 ## Class-side
 
+## Cog
+Cog is the name of the [virtual machine](#virtual-machine) used by Pharo. Cog is a virtual machine used by multiple programming languges such as Pharo, [Squeak](#squeak), Newspeak and Cuis.
+
 ## Context
 
 ## Debugger
 
 ## Dispatch
-Dispatch is a technique used in object-oriented programming to delegate behavior using polymorphism. The goal is to define a concept that will changed depending on the context. For example we can have an application with commands. When we need to execute the command, instead of creating a switch/case, each object will implément their own behavior on a method of the same name and we just need to call this method. 
+Dispatch is a technique used in object-oriented programming to delegate behavior using [polymorphism](#polymorphism). The goal is to define a concept that will changed depending on the context. For example we can have an application with commands. When we need to execute the command, instead of creating a switch/case, each object will implement their own behavior on a method of the same name and we just need to call this method. 
```

Comment:
```suggestion
Dispatch is a technique used in object-oriented programming to delegate behavior using [polymorphism](#polymorphism). The goal is to define a concept that will changed depending on the context. For example we can have an application with commands. When we need to execute the command, instead of creating a switch/case, each object will implement its own behaviour on a method of the same name and we just need to call this method. 
```